### DISCUSSION TITLE
Integration with password managers

### DISCUSSION THREAD

**Discussion by nguyenkims** [simple-login/app#412] [Feature Requests]
Bitwarden, 1Password, KeePass, etc

Moved from https://trello.com/c/D3qNPrF2/145-integration-with-password-managers

**Comment by developStorm** [comment]
Really wish we could partner with @1Password 🤞

**Comment by fily1212** [comment]
I hope Bitwarden, that is also open source

**Comment by WhamboMPS** [comment]
I believe this is a good idea. There are regular third-party reviews of password managers and it would be great to use the top-ranked tools as the candidates for integration. Examples:

https://www.pcmag.com/picks/the-best-password-managers
https://www.cnet.com/tech/services-and-software/best-password-manager/
https://www.tomsguide.com/us/best-password-managers,review-3785.html

**Comment by AedenThomas** [comment]
Any updates on this @nguyenkims

**Reply by nguyenkims** [reply]
We have contacted some password managers but there's no confirmation on the integration of SimpleLogin yet.

**Reply by UnreaI** [reply]
+1 Would really like to see Bitwarden/1pass integration! 🙏

**Reply by collide-vad** [reply]
1password and SimpleLogin would be great together :-)

**Comment by TheFou** [comment]
Hi,
FYI, 1Password partnered with Fastmail to offer this feature.
Let's now hope they are open to work with multiple partners, and there is no exclusivity...
Any update on your side @nguyenkims ?

**Reply by nguyenkims** [reply]
@Dj3ross we don't have any update for now.

**Comment by Vnzq4Qje3zzU** [comment]
This can be interesting: https://www.reddit.com/r/1Password/comments/scm9df/-/hu84nxd

> Of course, I'd be happy to answer your question! I'm delighted to hear you've had good experiences with our customer service team, they are the best, and I've heard great things about SimpleLogin's CS team as well.
>
> We keep a running list of email alias providers and SimpleLogin is at the top of that list. I can't promise anything, but know that we hear you and we would love to add more providers soon, [especially if they have a JMAP API](https://blog.1password.com/making-masked-email-with-jmap/). Stay tuned!

**Reply by paulverbeke** [reply]
Yes 1password will definitely prioritize alias services with a JMAP API implemented 
https://1password.community/discussion/comment/618181#Comment_618181

> JMAP was definitely one point that was attractive to us about Fastmail. <3

https://blog.1password.com/making-masked-email-with-jmap/

**Comment by mygrexit** [comment]
Have you guys heard anything from Bitwarden regarding a possible implementation?

**Reply by mygrexit** [reply]
nvm you just have been bought by proton :D

**Comment by Vnzq4Qje3zzU** [comment]
Bitwarden just started to implement and test something similar: 

https://www.reddit.com/r/Bitwarden/comments/u7t0bq/username_generator_released_for_web_vault_share/
https://bitwarden.com/help/generator/#generate-a-username

It's not a tight integration like 1Password and Fastmail but it already works for everyone with a custom domain with catch-all.

**Comment by Vnzq4Qje3zzU** [comment]
https://www.reddit.com/r/Bitwarden/comments/ucd9d2/comment/i6cbq3q/
https://www.reddit.com/r/Bitwarden/comments/ucd9d2/comment/i6dt54n/

It's coming!

@nguyenkims can you share details if you are working together with Bitwarden or will they probably create a standard which you will have to look after?

**Reply by nguyenkims** [reply]
@bx2KdaukwF Currently we don't have any news about this yet. We'll make sure to inform everyone when an integration is about to arrive :).

**Comment by bradenglund** [comment]
iOS app has recently added username generator to the keyboard - see https://github.com/simple-login/Simple-Login-iOS/issues/59.

When using the iOS Passwords feature to create a new login on a site (and to save into bitwarden) you can utilize the new keyboard to generate an alias on the spot.

This could be improved by making the creation tab of the keyboard customizable with you preferred domains, and url based alias prefixes as an option.


![IMG_1006](https://user-images.githubusercontent.com/28195153/166108720-16e5ecd2-1616-403a-8a14-fd4da93a55d8.PNG)

**Reply by ntnhon** [reply]
Now I get what you were saying in [#59](https://github.com/simple-login/Simple-Login-iOS/issues/59). Technically what you want to achieve is possible but I don't think it's really needed. For this particular use case, I think I'll make the Share Extension able to share a text, so that you can type out the URI and "share" it to open the Share Extension.

**Reply by ntnhon** [reply]
I created an [issue](https://github.com/simple-login/Simple-Login-iOS/issues/118) for this

**Reply by miguelarios** [reply]
When was this release? I don't have the iOS username generator. Only on browser extension.

**Comment by r-sw-eet** [comment]
Just started using simple login and it has much potential, but like the others say, it needs to work hand in hand with a password manager.
I use keepass2, but it is not very user friendly. 
There has been keeweb, i just found, but it is no longer maintained. https://github.com/keeweb/keeweb/issues/2022
How about forking the project and make your own integrations :)
Bitwarden and Onepassword are not good enough for me, although the look very handy, i dont like to save my passwords in the cloud (surely not on US servers), so i prefer offline data and sync the database myself.

**Reply by miguelarios** [reply]
> Bitwarden and Onepassword are not good enough for me, although the look very handy, i dont like to save my passwords in the cloud (surely not on US servers), so i prefer offline data and sync the database myself.

You can selfhost Bitwarden btw.

**Reply by EricGuic** [reply]
Yes, selfhost Vaultwarden on your own NAS or on a VPS you rent, and your quite safe ;-)

**Reply by sscotth** [reply]
> I use keepass2, but it is not very user friendly.
> There has been keeweb, i just found, but it is no longer maintained. https://github.com/keeweb/keeweb/issues/2022

[KeePassXC](https://github.com/keepassxreboot/keepassxc) is actively maintained

**Comment by Vnzq4Qje3zzU** [comment]
It just shipped on the Bitwarden's web vault, the browser extensions will probably receive it this week or in the next :smiley:

**Reply by asuman1179** [reply]
Hey wondering where in SimpleLogin do we find the API of our account?  Been looking around for it just not found anything just yet. Thanks

**Reply by MrMeeb** [reply]
Agreed - looks like custom API paths aren't support for SimpleLogin? I self-host mine and doesn't seem I can use it with Bitwarden. Am I missing something?

**Reply by 2fst4u** [reply]
In your simplelogin account, click on the little profile picture looking icon, not the regular three horizontal lines (hamburger) menu. Then there is a selection called "API keys".

**Reply by MrMeeb** [reply]
Yeah, I know where to find API keys. But there doesn't seem to be a way to change the API URL to use a self-hosted instance. I'm not really sure whether it's a SimpleLogin problem or a Bitwarden problem. Hopefully it comes in the future

**Reply by 2fst4u** [reply]
Ah apologies I misread. In fairness I struggled to find the APIs menu myself so at least that hint is there for posterity if anyone else is lost like I was.

**Comment by lavender-sys** [comment]
is there an update on this yet for 1Password?

**Reply by fm** [reply]
Considering they're now owned by Proton and they launched [Proton Pass](https://proton.me/pass) this week, I highly doubt it will happen. It could happen, but I highly doubt it.

**Reply by 2fst4u** [reply]
> Considering they're now owned by Proton and they launched [Proton Pass](https://proton.me/pass) this week, I highly doubt it will happen. It could happen, but I highly doubt it.

Bitwarden is already supported. It's not out of the realm of possibility.

**Reply by github-mmfna** [reply]
Maybe Proton doesn't feel it's in the best interest of their business however I suspect they have much to gain from such a partnership with 1P. Not only does it positively impact their cash flow by directly promoting and marketing SimpleLogin as a product, this will indirectly lead more 1P users to check out Proton as a result. ProtonPass is a competing product yes, but their other lines of business (VPN, Email, Drive) are wonderful supplements to 1P. No reason you can't have a subscription to both Proton and 1P, as many folks already do.

**Reply by olavinto** [reply]
I doubt that Proton will suddendly close off SimpleLogin. SL is open source just like all Proton apps are and basically anyone is free to add support via their API (also how Bitwarden connects to it) and this doesn't really have anything to do with Simple Login/Proton. Proton is also not driven simply by profit and self-interest so there is no reason why they would drop 3rd party integration support.

**Reply by ghost** [reply]
It will be better if @nguyenkims and @acasajus will comment

**Reply by JReming85** [reply]
Just because their parent company is a competitor doesn't mean they can't integrate with others.

My old company was bought up by a company in the space we were servicing in, and by law we were required to treat them the same as all the other companies we worked with.  Moreover, all data had to be sanitized before handing over since it had data on their competitors. (lawyers packets were happy each quarter).

**Comment by adriaandotcom** [comment]
For those who like 1Password support, let's ask them together.

Go to [support.1password.com/contact](https://support.1password.com/contact/).

| Click "I want to submit a feature request" | Click "Email" and enter message |
|----|----|
| ![2024-01-11-15h16-screenshot 2@2x](https://github.com/simple-login/app/assets/1079135/c91cc1b8-2156-431a-90f2-828318c7a72f) | ![2024-01-11-15h16-screenshot@2x](https://github.com/simple-login/app/assets/1079135/9a1206ec-e7ba-4b44-bad6-fd9053b034a9) |
